java/code/src/com/redhat/rhn/domain/contentmgmt/modulemd/ModulemdApi.java

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (c) 2020 SUSE LLC
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package com.redhat.rhn.domain.contentmgmt.modulemd;
+
+import com.google.gson.FieldNamingPolicy;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+import com.redhat.rhn.common.conf.Config;
+import com.redhat.rhn.common.conf.ConfigDefaults;
+import com.redhat.rhn.domain.channel.Channel;
+import com.redhat.rhn.domain.channel.Modules;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * modulemd Python API service
+ */
+public class ModulemdApi {
+
+    private static final String MOUNT_POINT_PATH = Config.get().getString(ConfigDefaults.MOUNT_POINT);
+    private static final String API_EXE = "mgr-libmod";
+    public static final Gson GSON =
+            new GsonBuilder().setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES).create();
+
+    /**
+     * Get all defined modules in a channel's metadata
+     *
+     * @param channel the modular channel
+     * @return a map of module name to module/stream objects
+     */
+    public Map<String, List<Module>> getAllModulesInChannel(Channel channel) {
```

Comment:
Is this method used?

Reply:
It's just a placeholder for one of the next steps mentioned here: https://github.com/SUSE/spacewalk/issues/10098

The corresponding `mgr-libmod` endpoint is already implemented.